src/lib.rs

```diff
@@ -2871,6 +2871,11 @@ impl Connection {
             self.error = Some(err);
         }
 
+        // When no packet was successfully processed close connection immediately.
+        if self.recv_count == 0 {
+            self.closed = true;
```

Comment:
Is it better to do at `drop_pkt_on_err()`?

Reply:
This is not really possible as we can't reference `self` when `drop_pkt_on_err()` is called (which is why it is a separate function rather than a `Connection` method).